### ISSUE TITLE
权重文件

### ISSUE THREAD

**Issue by LittleJJIU** [fyw1999/MovingDroneCrowd#13]
作者您好，请问能否提供训练的权重文件，我按照要求训练100轮后结果不佳，测试结果和论文相差较大

**Comment by fyw1999** [comment]
> 作者您好，请问能否提供训练的权重文件，我按照要求训练100轮后结果不佳，测试结果和论文相差较大

你好，权重已更新，建议训练时使用预训练的密度图估计模型，总batch_size设置为4，训练120个epoch试试

**Comment by LittleJJIU** [comment]
> > 作者您好，请问能否提供训练的权重文件，我按照要求训练100轮后结果不佳，测试结果和论文相差较大
> 
> 你好，权重已更新，建议训练时使用预训练的密度图估计模型，总batch_size设置为4，训练120个epoch试试

好的，非常感谢

**Comment by 15378695626** [comment]
> > > 作者您好，请问能否提供训练的权重文件，我按照要求训练100轮后结果不佳，测试结果和论文相差较大
> > 
> > 
> > 你好，权重已更新，建议训练时使用预训练的密度图估计模型，总batch_size设置为4，训练120个epoch试试
> 
> 好的，非常感谢

您好，采用这个总batch_size设置为4，训练120个epoch，结果任然与论文差距较大

**Comment by fyw1999** [comment]
> > > > 作者您好，请问能否提供训练的权重文件，我按照要求训练100轮后结果不佳，测试结果和论文相差较大
> > > 
> > > 
> > > 你好，权重已更新，建议训练时使用预训练的密度图估计模型，总batch_size设置为4，训练120个epoch试试
> > 
> > 
> > 好的，非常感谢
> 
> 您好，采用这个总batch_size设置为4，训练120个epoch，结果任然与论文差距较大

你怎么选取的权重呢，是选择的在验证集上seq_MAE最小的权重吗

**Comment by 15378695626** [comment]
> > > > > 作者您好，请问能否提供训练的权重文件，我按照要求训练100轮后结果不佳，测试结果和论文相差较大
> > > > 
> > > > 
> > > > 你好，权重已更新，建议训练时使用预训练的密度图估计模型，总batch_size设置为4，训练120个epoch试试
> > > 
> > > 
> > > 好的，非常感谢
> > 
> > 
> > 您好，采用这个总batch_size设置为4，训练120个epoch，结果任然与论文差距较大
> 
> 你怎么选取的权重呢，是选择的在验证集上seq_MAE最小的权重吗

是的，best_model.pt是按照在验证集上seq_MAE最小来选择的

**Comment by 15378695626** [comment]
是的，best-model.pt是按照验证集上的sql_MAE最小选择的



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2026年6月8日(周一) 晚上8:05
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [fyw1999/MovingDroneCrowd] 权重文件 (Issue #13)


fyw1999 left a comment (fyw1999/MovingDroneCrowd#13)
     
作者您好，请问能否提供训练的权重文件，我按照要求训练100轮后结果不佳，测试结果和论文相差较大
  
你好，权重已更新，建议训练时使用预训练的密度图估计模型，总batch_size设置为4，训练120个epoch试试
  
好的，非常感谢
  
您好，采用这个总batch_size设置为4，训练120个epoch，结果任然与论文差距较大
  
你怎么选取的权重呢，是选择的在验证集上seq_MAE最小的权重吗

**Comment by fyw1999** [comment]
The reproduction was conducted respectively under the global batch sizes of 4 and 8, and the results are as follows. Note that, to speed up the test, set `skip_flag` to `True`. As a result, the MIAE and MIOE results are inaccurate.
When global batch sizes is 8:

<img width="831" height="870" alt="Image" src="https://github.com/user-attachments/assets/43edb797-4335-4b5c-bbbb-919764daec0e" />

When global batch sizes is 4:

<img width="819" height="865" alt="Image" src="https://github.com/user-attachments/assets/175de994-6a00-4ef5-9838-212c30dfd0ec" />

When the global batch size is 8, some metrics (such as overall MAE) perform better than what was reported in the ICCV paper. The replicated results also align with the state-of-the-art performance of the competing algorithms at that time. SDNet is a weakly-supervised algorithm, which inherently has training instability and cannot guarantee that the results of each reproduction will be exactly the same. However, it can ensure that the SOTA performance of the comparison in the original paper. This instability is also one of the reasons for proposing the subsequent GD3A. If you want to reproduce the results of SDNet, it is best to ensure that the global batch size is 8 and use the pre-trained global counter.
The parameters for this reproduction are set as follows:
```python
import os
from easydict import EasyDict as edict
import time
# init
__C = edict()
cfg = __C

#------------------------------TRAIN------------------------
__C.SEED = 3035  # random seed,  for reproduction
__C.MODEL = "SDNet"  # model selection: GD3A, SDNet
__C.NAME = 'sdnet_test'  # name of the experiment
__C.DATASET = 'MovingDroneCrowd'       # dataset selection:  HT21, SENSE
__C.encoder = "VGG16_FPN"
__C.RESUME = False # continue training
__C.RESUME_PATH = ''
__C.PRE_TRAINED_MODEL = ''
__C.GPU_ID = '0,1,2,4'

#   for GD3A
__C.global_counter = "STEERER"
__C.pre_trained_global_counter = ""
__C.sinkhorn_iterations = 100
__C.down_scale = 8
__C.ROI_RADIUS = 2.
__C.filter_thre = 0.005
__C.matched_thre = 0.1
__C.top_k = 9
__C.LR_Thre = 1e-4

# for SDNet
__C.PRE_TRAIN_COUNTER = 'pre_train_model/SDNet_pre_trained_counter_MDC_VGG16_FPN_ep_200_downscale_16.pth'
__C.cross_attn_embed_dim = 256
__C.cross_attn_num_heads = 4
__C.mlp_ratio = 4
__C.cross_attn_depth = 2

# for both
__C.FEATURE_DIM = 256
__C.LR_Base = 1e-5  # learning rate
__C.WEIGHT_DECAY = 1e-6

__C.MAX_EPOCH = 120
__C.VAL_INTERVAL = 10
__C.START_VAL = 20
__C.PRINT_FREQ = 20
# print
now = time.strftime("%m-%d_%H-%M", time.localtime())

__C.EXP_NAME = now \
    + '_' + __C.DATASET \
    + '_' + str(__C.LR_Base) \
    + '_' + __C.NAME

__C.EXP_PATH = os.path.join('./exp', __C.DATASET)  # the path of logs, checkpoints, and current codes

os.makedirs(__C.EXP_PATH, exist_ok=True)
```
The settings for the relevant parameters of the dataset are as follows:
```python
from easydict import EasyDict as edict


# init
__C_MDC = edict()

cfg_data = __C_MDC

__C_MDC.TRAIN_SIZE = (768, 1024) # (848,1536) #
__C_MDC.TRAINING_MAX_LONG = 2560
__C_MDC.TRAINING_MAX_SHORT = 1440
__C_MDC.TEST_MAX_LONG = 1920
__C_MDC.TEST_MAX_SHORT = 1080
__C_MDC.DATA_PATH = '/data1/fyw/datasets/counting/MovingDroneCrowd++/'
__C_MDC.TRAIN_LST = 'MDC_train.txt'
__C_MDC.VAL_LST =  'MDC_val.txt'
__C_MDC.TEST_LST =  'MDC_test.txt'

__C_MDC.MEAN_STD = (
    [117/255., 110/255., 105/255.], [67.10/255., 65.45/255., 66.23/255.]
    # [0.4368163,  0.42558577, 0.41953506], [0.25696197, 0.25339108, 0.25411048]
)

__C_MDC.DEN_FACTOR = 200.

__C_MDC.TRAIN_BATCH_SIZE = 2 #  img pairs
__C_MDC.TRAIN_FRAME_INTERVALS=(3, 8)
__C_MDC.VAL_FRAME_INTERVALS = 4
__C_MDC.VAL_BATCH_SIZE = 1 # must be 1
```
Tips: Noted that during the reproduction process, both configurations achieved the best results on the validation set at the 120th epoch (best_model.pth). If you still cannot reproduce the results, it is recommended to further increase the number of training epochs (e.g., to 150) while ensuring that the global batch size is 8 and using the pre-trained global counter to further reduce the impact of instability training on the final results.
